tests/JavaTest.java

```diff
@@ -121,6 +135,17 @@ public static void main(String[] args) {
         // the mana field should retain its default value
         TestEq(monster.mutateMana((short)10), false);
         TestEq(monster.mana(), (short)150);
+		
+		// Accessing a vector of sorted by the key tables
+        Monster[] monsters = { monster.testarrayoftables(0), monster.testarrayoftables(1), monster.testarrayoftables(2) };
```

Comment:
What if it is a very long vector? Having to copy the offsets out of the vector seems really unpractical, it be better if `lookupByKey` directly worked on the vector in the buffer.